main.py

```diff
@@ -87,6 +87,14 @@ def stored_flows_list(self, dpid):
         """Ordered list of all stored flows given a dpid."""
         return itertools.chain(*list(self.stored_flows[dpid].values()))
 
+    def stored_flows_by_state(self, dpid, state):
```

Comment:
This kind of filtering here we can delegate and perform directly on a DB later as we move to a new back-end later on.